### PR TITLE
feat!: add testnet cmd

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #563 
Ref: [#9246](https://github.com/cosmos/cosmos-sdk/pull/9246)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by ryanchristo — APPROVED** [review]
tACK. Cross-referenced with the implementation in cosmos-sdk. Looks good! The only request would be to be to add `.testnets` to `.gitignore` but pre-approving.

Also, one issue with running `make localnet-start` but unrelated to the implementation. It seems to be an issue using the latest `libc`, which is available on Ubuntu 21.10.

File: Makefile [hunk 1]
```diff
@@ -392,16 +393,16 @@ proto-update-deps:
 ###                                Localnet                                 ###
 ###############################################################################
 
-# Run a 4-node testnet locally
+# Run a 4-node testnet locally via docker compose
 localnet-start: build-linux localnet-stop
-	$(if $(shell docker inspect -f '{{ .Id }}' tendermint/xrnnode 2>/dev/null),$(info found image tendermint/xrnnode),$(MAKE) -C contrib/images xrnnode)
-	if ! [ -f build/node0/simd/config/genesis.json ]; then docker run --rm \
+	$(if $(shell $(DOCKER) inspect -f '{{ .Id }}' regenledger/regen-env 2>/dev/null),$(info found image regenledger/regen-env),$(MAKE) -C contrib/images regen-env)
+	if ! test -f build/node0/regen/config/genesis.json; then $(DOCKER) run --rm \
```
[the remark this answers is not in the document]
Comment (aleem1314) [reply Makefile:399]: Updating the base image to `ubuntu:21.04` solves the issue.

[the remark this answers is not in the document]
Comment (anilCSE) [reply Makefile:399]: I am getting the following error on Ubuntu 18.04.
```sh
Creating network "regen-ledger_localnet" with driver "bridge"
ERROR: Pool overlaps with other one on this address space
Makefile:398: recipe for target 'localnet-start' failed
make: *** [localnet-start] Error 1
```

**Review by anilCSE — APPROVED** [review]

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleem1314) [reply Makefile:399]: > ```sh
> Creating network "regen-ledger_localnet" with driver "bridge"
> ERROR: Pool overlaps with other one on this address space
> Makefile:398: recipe for target 'localnet-start' failed
> make: *** [localnet-start] Error 1
> ```

I'm also using 18.04 and it is working fine. I think `192.168.10.0/16` subnet is already in use. Can u try [this](https://github.com/maxking/docker-mailman/issues/85#issuecomment-317283012).

[the remark this answers is not in the document]
Comment (anilCSE) [reply Makefile:399]: Thanks, this worked

**Pull request merged**